lib/features/academic_calendar/utils/counter_digits.dart

```diff
@@ -2,18 +2,24 @@ import "dart:math";
 
 import "../../../config/ui_config.dart";
 import "../../../utils/datetime_utils.dart";
+import "../model/academic_calendar_extensions.dart";
 import "../repository/academic_calendar_repo.dart";
 
 enum Digit { first, second, third }
 
 extension DaysLeftStringConverter on AcademicCalendar {
-  String get daysLeftFromNowString => max(
-        _default,
-        data?.examSessionStartDate.daysLeftFromNow ?? _default,
-      ).toString().padLeft(
-            Digit.values.length,
-            _default.toString(),
-          );
+  String get daysLeftFromNowString {
+    final calendarData = this.AcademicCalendarData;
+    final daysLeft = calendarData != null
+        ? (calendarData.isHolidays()
+            ? data?.semesterStartDate.daysLeftFromNow
```

Comment:
So this basically here have only two modes/cases? Am I right? Shouldn't there be three cases as in you PR description? I think you don't have the case when we are in semester and we want to see days until the START of the exam session? Or am I missing smth?

Reply:
Sorry for that. Of course you're right here.  I think's fixed rn.